### PR TITLE
Convert to maven project and add nashorn-core dependency

### PR DESCRIPTION
**Pull request by jvierling**
The pull request converts the project to a maven project and includes the `nashorn-core` dependency that provides a javascript engine. The mmj2 executable JAR file can be built with the command `mvn clean package`.

### CODE REVIEW

File: src/main/java/mmj/gmff/EscapePair.java [hunk 1]
```diff
@@ -36,7 +36,7 @@
  */
 
 public class EscapePair {
-    public int num;
+    public int num ;
```
Comment (benjub) [inline src/main/java/mmj/gmff/EscapePair.java]: why the added space ?

File: build.xml [hunk 2]
```diff
@@ -1,74 +0,0 @@
...
-                <exclude name="**/*.java"/>
-            </fileset>
-        </copy>
-    </target>
-    <target name="clean">
-        <delete dir="classes"/>
-        <delete dir="doc/javadoc"/>
-    </target>
-    <target depends="clean" name="cleanall"/>
-    <target depends="init" name="build-project">
```
Comment (digama0) [inline build.xml:26]: Uh, are you sure you want to delete the ant build script? I don't think you have replicated all the targets here. Ideally the ant build would continue to work, with only some path changes.

Comment (tirix) [reply build.xml:26]: Is it possible to revert [just that commit](https://github.com/digama0/mmj2/pull/65/commits/e1ab7fbcbd2eac376e9bda8fdfe28f72a95a1f14)?

Comment (jvierling) [reply build.xml:26]: I'll remove the commit and try to make the ant script work with the new path.

**Comment by jvierling** [comment]
I have cleaned the commits according to the review suggestions. Let me know if there is anything else that needs to be fixed.

**Comment by digama0** [comment]
Can you add build documentation to the README? Also a CI action which builds the project would be nice but that's probably a separate PR.

**Comment by jvierling** [comment]
I have added the build instructions to `INSTALL.md`, since the other build instructions are located in that file as well. The Github CI Action sets up the submodules and builds the JAR using the `mvn package` command.

File: INSTALL.md [hunk 3]
```diff
@@ -319,3 +319,27 @@ can legally use it).
...
+
+~~~
+mvn package
+~~~
+
+The JAR file `mmj2-2.5.2-SNAPSHOT-jar-with-dependencies.jar` will be
+created with in the directory `mmj2/target` and can be run using
+
+~~~
+java -jar target/mmj2-2.5.2-SNAPSHOT-jar-with-dependencies.jar <mmj2-options>
```
Comment (digama0) [inline INSTALL.md:344]: The previous workflow has the jar file appearing in `mmj2jar/mmj2.jar`, and the scripts in the folder also assume that this jar file is correct and up to date. Perhaps the build instructions should suggest a `mv target/mmj2-2.5.2-SNAPSHOT-jar-with-dependencies.jar mmj2jar/mmj2.jar` step, or `mvn package` should do that itself. (I'm also not a fan of baking in version info into the file name, as that makes scripts break frequently.)

Comment (jvierling) [reply INSTALL.md]: The build instructions now mention the copying step necessary to invoke the `mmj2` as described in the quick start guide.

File: INSTALL.md [hunk 4]
```diff
@@ -337,9 +337,17 @@ After that, build the executable JAR file using the following command:
...
 ~~~
 
-The JAR file `mmj2-2.5.2-SNAPSHOT-jar-with-dependencies.jar` will be
-created with in the directory `mmj2/target` and can be run using
+The JAR file `mmj2-${project.version}-jar-with-dependencies.jar` will be
+created with in the directory `mmj2/target` and can be run directly using
 
 ~~~
-java -jar target/mmj2-2.5.2-SNAPSHOT-jar-with-dependencies.jar <mmj2-options>
+java -jar target/mmj2-${project.version}-jar-with-dependencies.jar <mmj2-options>
```
Comment (digama0) [inline INSTALL.md:344]: Is this `${project.version}` syntax expected to be expanded by the reader to whatever the latest is? If so, you should give an example and point to the place where they can find out what actual value to put in there.

Comment (jvierling) [reply INSTALL.md:346]: I have modified the commits to include an explanation of the `${project.version}` syntax and where to find the version number that `${project.version}` stands for.

**Pull request merged**